lib/internal/fs/recursive_watch.js

```diff
@@ -173,22 +142,30 @@ class FSWatcher extends EventEmitter {
       return;
     }
 
-    const { watchFile } = lazyLoadFsSync();
-    const existingStat = this.#files.get(file);
+    const { watch, statSync } = lazyLoadFsSync();
+
+    if (this.#files.has(file)) {
+      return;
+    }
+
+    {
+      const existingStat = statSync(file);
```

Comment:
My only concern is that, for several places, we make `statSync` calls twice. One in the parent caller and one in this function.